src/js/modules/media/selectors.js

```diff
@@ -0,0 +1,65 @@
+import { createSelector } from 'reselect';
+import { getArticles } from "../articles/selectors";
+import { objectFilter } from "../../utils";
+import { getUsers} from "../users/selectors";
+
+Object.filter = objectFilter;
+
+const getMedia = state => state.media.media;
+
+export const matchMediaToArticles = createSelector(
+  [ getMedia, getArticles ],
+  ( medias, articles ) => {
+    const newArticleObject = {};
+    Object.keys(articles).map((article) => {
+      const articleObject = articles[article];
+      const matchingMedia = Object.filter(medias, media => {
```

Comment:
Media is plural already. Maybe make the current `media` parameter `mediaObject`.